modules/navier_stokes/doc/content/source/postprocessors/INSElementIntegralEnergyAdvection.md

```diff
@@ -0,0 +1,40 @@
+# INSElementIntegralEnergyAdvection
+
+!syntax description /Postprocessors/INSElementIntegralEnergyAdvection
+
+This class performs global integration of the energy advection term defined in
+[INSADEnergyAdvection.md], which is defined by
+
+\begin{equation}
+\rho c_p \vec u \cdot \nabla T
+\end{equation}
+
+where $\rho$ is the density, $c_p$ is the
+specific heat capacity,
+$\vec u$ is the velocity, and $\nabla T$ is the temperature gradient. AD
+instantiations of this object are denoted by
+`INSADElementIntegralEnergyAdvection`, hand-coded by
+`INSElementIntegralEnergyAdvection`. The former retrieves the density and
+specific heat capacity as `ADReal` material properties while the former
+retrieves the same properties as `Real`.
+
```

Comment:
you should add a note to say why this is more appropriate than doing surface based balances

Reply:
:+1: